pkg/descheduler/policyconfig.go

```diff
@@ -45,26 +46,125 @@ func LoadPolicyConfig(policyConfigFile string, client clientset.Interface, regis
 		return nil, fmt.Errorf("failed to read policy config file %q: %+v", policyConfigFile, err)
 	}
 
+	internalPolicy, err := decode(policyConfigFile, policy, client, registry)
+	if err != nil {
+		return nil, err
+	}
+	return internalPolicy, nil
+}
+
+func decode(policyConfigFile string, policy []byte, client clientset.Interface, registry pluginregistry.Registry) (*api.DeschedulerPolicy, error) {
 	versionedPolicy := &v1alpha1.DeschedulerPolicy{}
+	internalPolicy := &api.DeschedulerPolicy{}
+	var err error
 
-	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion)
+	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion, v1alpha2.SchemeGroupVersion, api.SchemeGroupVersion)
 	if err := runtime.DecodeInto(decoder, policy, versionedPolicy); err != nil {
-		return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
+		if err.Error() == "converting (v1alpha2.DeschedulerPolicy) to (v1alpha1.DeschedulerPolicy): unknown conversion" {
+			klog.V(1).InfoS("Tried reading v1alpha2.DeschedulerPolicy and failed. Trying legacy conversion now.")
+		} else {
+			return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
+		}
 	}
-
-	// Build profiles
-	internalPolicy, err := V1alpha1ToInternal(client, versionedPolicy, registry)
-	if err != nil {
-		return nil, fmt.Errorf("failed converting versioned policy to internal policy version: %v", err)
+	if versionedPolicy.APIVersion == "descheduler/v1alpha1" {
+		// Build profiles
+		internalPolicy, err = V1alpha1ToInternal(client, versionedPolicy, registry)
```

Comment:
maybe I'm wrong but I thought with conversion funcs set up we should just be able to pass either one to decode and let the conversion functions handle it. seems odd that we still have that extra layer here. Either way, I think the `V1Alpha1ToInternal` function should be in the api package, not here (if we do need it)

Reply:
So, we have most of the auto generated conversions working for v1alpha2 to api because they are very similar (with exception to the runtime.Objects/RawExtensions args, which made us do some custom code).

v1alpha1 and api are very different, and none of it can be auto-generated.

Also we have some very custom ways to set defaults, and specially, setting thresholdPriorities. Right now we depend on having a client passed around. 

The other thing is strategyParamsToPluginArgs map, that lives in strategy_migration.go, and we cant import descheduler.go in v1alpha1, as descheduler.go imports v1alpha1. 

So moving this function to v1alpha1 would require some serious moving around. I think it is easier to keep it in the policyconfig.go file since it is temporary, and also very custom with all the conversion from strategies to profiles.

Reply:
I tried making V1alpha1ToInternal just become a normal Convert_v1alpha1_DeschedulerPolicy_To_api_DeschedulerPolicy in v1alpha1/conversion.go, but gave up mid tryin it :sweat_smile:

Reply:
Ok, so, I tried moving things around to avoid import cycles. We still have V1Alpha1ToInternal func, I don't want to create another client inside a default `Convert_v1alpha1_DeschedulerPolicy_To_api_DeschedulerPolicy` func... so better to have this as something that we call so we can pass registries+clients and stuff around